packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -3848,7 +3848,9 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
         if (paintBounds.bottom <= box.top) {
           break;
         }
-        if (paintBounds.contains(Offset(box.left, box.top)) || paintBounds.contains(Offset(box.right, box.bottom))) {
+        // Include any rect with overlap with paintBounds.
+        if ((paintBounds.left <= box.right || paintBounds.right >= box.left) &&
+            paintBounds.top <= box.bottom) {
```

Comment:
not sure if i follow this logic. can you put more explanation here?

Reply:
Before, it only checked the corners, so it would miss any box which is taller than the paintBounds. Now, we should include any box with any overlap at all with paintBounds. It's basically `Rect.overlaps`, but I removed one check since it was already covered a little above the highlighted snippet. And also, I found a test that depended on the overlap check being inclusive, while `Rect.overlaps` requires a non-zero area of overlap. (The existing test expected that a box that was to the left of paintBounds and only touching on the edge would be included).

Reply:
> And also, I found a test that depended on the overlap check being inclusive

Can you clarify if the expected behavior should include the case with just edge touching?

Reply:
I included it to not change the existing test.

Reply:
do you also need 2 conditions for top/bottom?

Reply:
For checking 2 rects intersecting with each other, i normally do this: 

```
! (r1.right < r2.left || r1.left > r2.right || r1.top > r2.bottom || r1.bottom < r2.top)
```

Or can remove the negate with de morgan's, but that's less clear to reason about.

Reply:
Look at the latest version of the commit, we are commenting on a version which I changed. I think I have the same thing with demorgan's applied. But the last condition for top/bottom is earlier on in the loop, it does a `break` when the rect is below the text box.

Reply:
Can you add some comments explaining this logic? (e.g. this is effectively checking for rect intersection, etc). This will help future readers to understand.